crates/tauri-utils/src/config/security.rs

```diff
@@ -0,0 +1,783 @@
+  pub cross_origin_opener_policy: Option<HeaderSource>,
+  /// The HTTP Cross-Origin-Resource-Policy response header conveys a desire that the
+  /// browser blocks no-cors cross-origin/cross-site requests to the given resource.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Resource-Policy>
+  #[serde(rename = "Cross-Origin-Resource-Policy")]
+  pub cross_origin_resource_policy: Option<HeaderSource>,
+  /// The HTTP Permissions-Policy header provides a mechanism to allow and deny the
+  /// use of browser features in a document or within any \<iframe\> elements in the document.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Permissions-Policy>
+  #[serde(rename = "Permissions-Policy")]
+  pub permissions_policy: Option<HeaderSource>,
+  /// The Timing-Allow-Origin response header specifies origins that are allowed to see values
+  /// of attributes retrieved via features of the Resource Timing API, which would otherwise be
+  /// reported as zero due to cross-origin restrictions.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Timing-Allow-Origin>
+  #[serde(rename = "Timing-Allow-Origin")]
+  pub timing_allow_origin: Option<HeaderSource>,
+  /// The X-Content-Type-Options response HTTP header is a marker used by the server to indicate
+  /// that the MIME types advertised in the Content-Type headers should be followed and not be
+  /// changed. The header allows you to avoid MIME type sniffing by saying that the MIME types
+  /// are deliberately configured.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options>
+  #[serde(rename = "X-Content-Type-Options")]
+  pub x_content_type_options: Option<HeaderSource>,
+  /// A custom header field Tauri-Custom-Header, don't use it.
+  /// Remember to set Access-Control-Expose-Headers accordingly
+  ///
+  /// **NOT INTENDED FOR PRODUCTION USE**
+  #[serde(rename = "Tauri-Custom-Header")]
+  pub tauri_custom_header: Option<HeaderSource>,
+}
+
+impl HeaderConfig {
+  /// creates a new header config
+  pub fn new() -> Self {
+    HeaderConfig {
+      access_control_allow_credentials: None,
+      access_control_allow_methods: None,
+      access_control_allow_headers: None,
+      access_control_expose_headers: None,
+      access_control_max_age: None,
+      cross_origin_embedder_policy: None,
+      cross_origin_opener_policy: None,
+      cross_origin_resource_policy: None,
+      permissions_policy: None,
+      timing_allow_origin: None,
+      x_content_type_options: None,
+      tauri_custom_header: None,
+    }
+  }
+}
+
+impl Display for HeaderConfig {
+  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+
+        write!(f,"Access-Control-Allow-Credentials: ")?;
+        match &self.access_control_allow_credentials {
+          Some(value) => writeln!(f, "{}", value)?,
+          None => writeln!(f, "null")?,
+        };
```

Comment:
This also can be simplified into
```suggestion
        writeln!(f,"Access-Control-Allow-Credentials: {}", self.access_control_allow_credentials .unwrap_or("null"))?;
```